src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,87 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { Todo } from '../../types/Todo';
+import { useAppSelector } from '../../app/hooks';
+import { Loader } from '../Loader';
+import { getTodos } from '../../api';
+import { actions as todosAction } from '../../features/todos';
+
+enum Status {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
 
 export const TodoList: React.FC = () => {
+  const todos = useAppSelector((state) => state.todos);
+  const { query, status } = useAppSelector((state) => state.filter);
+  const [isError, setIsError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setIsError(false);
+    setIsLoading(true);
+
+    getTodos()
+      .then((todosList) => dispatch(todosAction.addTodos(todosList)))
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    const newTodos = query
+      ? [...todos].filter((todo) => todo.title.toLowerCase().includes(query))
+      : [...todos];
+
+    switch (status) {
+      case Status.active:
+        return newTodos.filter((todo) => !todo.completed);
+
+      case Status.completed:
+        return newTodos.filter((todo) => todo.completed);
+
+      case Status.all:
+        return newTodos;
+
+      default:
+        return newTodos;
```

Comment:
```suggestion
      case Status.all:
      default:
        return newTodos;
```
you may join a few cases if you do the same things